### PR TITLE
Spec: add version to title

### PR DESCRIPTION
**Pull request by chalin**
Hi @open-telemetry/specs-approvers: before setting up a script to automatically fetch the latest version of the spec (#2387), I wanted to ensure that we agreed on how the version will be displayed. This PR is such a proposal, it:

- Adds the spec version to:
  - The Spec landing page title -- see the preview
  - The `linkTitle` so that the version appears in the:
    - Left sidenav
    - Breadcrumbs, and hence in every spec page
- Contributes to #2387

**Preview**: https://deploy-preview-2666--opentelemetry.netlify.app/docs/reference/specification/

### Screenshots

> Screen Shot 2023-05-07 at 15 05 08

/cc @open-telemetry/specs-approvers @tigrannajaryan

### CODE REVIEW

**Comment by cartermp** [comment]
I would say this looks right to me, but I'll defer to the spec approvers.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@open-telemetry/specs-approvers please take a look!

File: scripts/content-modules/adjust-pages.pl
Comment (tigrannajaryan) [inline scripts/content-modules/adjust-pages.pl:19]: This is manually updated for now, right?

**Review by tigrannajaryan — APPROVED** [review]
LGTM, thanks.

File: scripts/content-modules/adjust-pages.pl
Comment (chalin) [reply scripts/content-modules/adjust-pages.pl:19]: @tigrannajaryan - right, but I'll setup the auto-update in my next spec-related PR -- it's one of the tasks of #2387.

Comment (chalin) [reply scripts/content-modules/adjust-pages.pl:19]: Here's the PR:

- #2674

**Pull request merged**
